### DISCUSSION TITLE
Where do you want to go?

### DISCUSSION THREAD

**Discussion by Corwii** [pyinstaller/pyinstaller#6811] [Help]
For the life of me I cannot figure this out. When I use Pyinstaller and it starts importing my modules, it gets to importing future.moves.dbm and stops to ask "Where do you want to go?" I can't figure out what to do here, and keyboard interrupting out of this cancels the rest of the build. I've tried posting to reddit, no one helped, tried reaching out to the devs of future, no response, so this is my last resort to solve this issue is to come straight to pyinstaller community for help.

I edited the build_main file of pyinstaller to print out the packages as it's importing them, it stops at future.moves.dbm, but there's many other future modules that it does this with, like pyinstaller has issues with future? I'm not sure what else to do here. I can add a list of future modules to exclude and bypass this but my app seems to freeze and loop, like it's running overtop itself over and over. So I'm just assuming it's because I can't get future to fully import, which pyglet uses.

![AA](https://user-images.githubusercontent.com/105116429/167258508-1fbf07ef-b5bc-48dd-a90a-0e298fd96102.PNG)
.

**Comment by rokm** [comment]
Can you provide a minimal example that triggers this issue?

Otherwise, you should start by figuring out where the string `Where do you want to go?` (and presumably subsequent `input()` call) actually comes from. For example install `pss` and search for the string in your code tree and in the site-packages you use.

Is it from *your* code? In which case, do you have a top-level module whose name shadows some standard python module and gets loaded instead? EDIT: also, what is the name of your entry-point script?

**Reply by Corwii** [reply]
Here's the list of future modules that seem to be asking this question during the import module phase

`['future.moves.dbm','future.moves','future.moves.test','dbm','future.backports','future.backports.email','future.backports.http','future.backports.urllib','future.moves.html']`

If I list these like this and exclude them from pyinstaller, it will finish compiling but like I said, the app starts but keeps looping and running over itself, so I'm not sure if this is an issue with excluding those modules, since pyglet depends on future, or if it's an issue with freezing. And I get this when I try to install pss.

Collecting pss
  Using cached pss-1.43.tar.gz (26 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [8 lines of output]
      C:\Python 3.8\lib\site-packages\setuptools\_distutils\dist.py:275: UserWarning: Unknown distribution option: 'entry_points'
        warnings.warn(msg)
      usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
         or: setup.py --help [cmd1 cmd2 ...]
         or: setup.py --help-commands
         or: setup.py cmd --help

      error: invalid command 'egg_info'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

Forgot to mention entry script is just  game.py

**Reply by rokm** [reply]
For `pss`, you'll probably need to upgrade `setuptools` and perhaps also `pip`. But you can search the sources using any method you want, really...

I'm almost certain that `future` is red herring here, because none of its modules contains that message.

 If you Ctrl+C during the `input()` (while waiting for the answer to "Where do you want to go?"), what does the traceback (presumably for `KeyboardInterrupt` exception) look like? Where does it originate from?

**Reply by rokm** [reply]
Do you have any other .py files next to your `game.py`?

**Comment by Corwii** [comment]
So when I keyboard Interrupt out it just says this.

Where do you want to go?>: Aborted by user request.

Makes sense to me that it's something else, I'm currently trying to do a
notepad++ sweep of a whole bunch of files containing that string to find
where it's coming from. Thank you for taking time to assist me. I'll update
you soon if I figure it out lol

**Comment by Corwii** [comment]
[image: Where do you want to go.PNG]

Seems to be taking place inside the __pycache__ folder,
test.cpython-38.pyc. This folder is created by pyinstaller from what I
understand. So this would be a pyinstaller issue.

**Reply by rokm** [reply]
> Seems to be taking place inside the __pycache__ folder, test.cpython-38.pyc. This folder is created by pyinstaller from what I understand. So this would be a pyinstaller issue.

Maybe in an alternate reality, but not in this one...

But by the sound of it, you have a corresponding `test.py` file lying around, and it's probably imported by `future.moves`. Either that, or you used to have a `test.py`, but now all that's left behind is the byte-compiled module, and *that* is what's being picked up.

Suppose if you open python interpreter, and run `import future.moves`, you get the same question? In that case, remove the `__pycache__/test.cpython-38.pyc` (or the whole `__pycache__` directory). EDIT: and remove/rename the `test.py` file if it is still around...

**Comment by Corwii** [comment]
Yes so this is solved, I found the test file in question. Definitely not
future.moves, I can import it just fine. Thank you again for your
assistance! Now onto the next issues lol.
